### ISSUE TITLE
Dark Mode Toggle Placement

### ISSUE THREAD

**Issue by kjellr** [WordPress/twentytwentyone#784]
I'd like revisit discussion about the toggle placement from WordPress/twentytwentyone-dark-mode#1. Currently, the dark mode toggle is placed here: 

![image](https://user-images.githubusercontent.com/1202812/97740888-4478bc80-1ab8-11eb-896b-da98039f3ec9.png)

I have a couple concerns with its current placement: 

## Prominence of the Button

This is a very strong visual element, and it is located above all other site content. Unless there's a prominent site logo, this button will likely be the first thing a user's eye is drawn to in the header. From a hierarchy perspective, I don't think that makes much sense: while we want to make this toggle feature available, it shouldn't be more important visually than the site title. 

This is especially an issue on mobile, where it takes up a larger proportion of the screen: 

![Screen Shot 2020-10-30 at 2 25 05 PM](https://user-images.githubusercontent.com/1202812/97743216-b9012a80-1abb-11eb-95ad-9a7e2c355cb7.png)

Does this need to exist in the header? I do not have data around this, but I'd guess that most users will not make use of this button at all. Does this need to be in the main header if that's the case?

## Position of the Button

Assuming this button does remain in the header, the current placement feels imbalanced. The header forms a weird "N" shape: 

![image](https://user-images.githubusercontent.com/1202812/97741977-0381a780-1aba-11eb-9e5c-35e1eec3ba2f.png)

If we do place the button in the header, I'd suggest we work to balance this shape out visually. Even something like this helps: 

![image](https://user-images.githubusercontent.com/1202812/97742094-362ba000-1aba-11eb-8eca-881ddbcbec89.png)

Moving it below the menu there would also improve the mobile situation, allowing the close button to once again appear in the same position as the open menu button. Currently, [they are in different spots](https://cloudup.com/cp1h9lIA1qE). 

It would be even more balanced to do something like this (without placing it in the actual `nav` element of course): 

![image2](https://user-images.githubusercontent.com/1202812/97742165-4a6f9d00-1aba-11eb-816e-6de12b4fae9a.png)

Are there concerns with this approach? Does anyone have any other ideas that I've missed?

**Comment by carolinan** [comment]
I still have accessibility concerns for the placement of the button in the right corner, because focus moves from the upper left skip link to the bottom right. This causes problems if you are zooming in or using a magnifier.

When zoomed in to say [400%](https://www.w3.org/WAI/WCAG21/Understanding/reflow.html), the fixed position means that the button covers a large part of the content.

This image is reduced in size but shows the problem:
![button-covers-content](https://user-images.githubusercontent.com/7422055/98090755-c126e480-1e84-11eb-9536-8c501301da2a.png)

**Comment by carolinan** [comment]
I think the experience would be improved if the button was not visible when scrolling.
That is a common pattern for example for go to top links.

**Comment by kjellr** [comment]
> I think the experience would be improved if the button was not visible when scrolling.
> That is a common pattern for example for go to top links.

I agree, that would be great to try.

**Comment by kjellr** [comment]
Additionally, I'm seeing some issues with the toggle button's placement and appearance in the editor. We can split this into its own issue if we'd like, but posting here for now: 

![dark-mode](https://user-images.githubusercontent.com/1202812/98244879-1a9f1a00-1f3e-11eb-9f29-2ef069d1d4c9.gif)

(I'm not 100% sure we need to include the button here in the first place... I see how it is helpful from a preview perspective, but it's clearly new functionality, not content. I'd expected the editor styles would just respect the same toggle setting I'd chosen on the front end.)

**Comment by aristath** [comment]
> I think the experience would be improved if the button was not visible when scrolling.

Done in #798 

> I'm not 100% sure we need to include the button here in the first place... I see how it is helpful from a preview perspective, but it's clearly new functionality, not content. I'd expected the editor styles would just respect the same toggle setting I'd chosen on the front end.

PR in #799

**Comment by carolinan** [comment]
Can we place the toggle last in the focus order?
Because I think moving the focus across the page from the top left to the bottom right is worse.

**Comment by aristath** [comment]
> Can we place the toggle last in the focus order?

PR: https://github.com/WordPress/twentytwentyone/pull/817

**Comment by carolinan** [comment]
@kjellr  @melchoyce We have not tried using a full width button fixed at the bottom of the page on mobiles, other than that I believe everything has been addressed and this can be closed?

**Comment by kjellr** [comment]
Yeah, I think the solution we have works ok for now. No need to try the full-width at the bottom. 👍 

Thanks!
